crates/router/src/connector/nuvei/transformers.rs

```diff
@@ -362,50 +418,13 @@ impl<F>
         ),
     ) -> Result<Self, Self::Error> {
         let item = data.0;
-        let session_token = data.1;
-        if session_token.is_empty() {
-            return Err(errors::ConnectorError::MissingRequiredField {
-                field_name: "session_token",
-            }
-            .into());
-        }
-        let connector_meta: NuveiAuthType = NuveiAuthType::try_from(&item.connector_auth_type)?;
-        let merchant_id = connector_meta.merchant_id;
-        let merchant_site_id = connector_meta.merchant_site_id;
-        let client_request_id = item.attempt_id.clone();
-        let time_stamp = date_time::date_as_yyyymmddhhmmss()
-            .into_report()
-            .change_context(errors::ConnectorError::RequestEncodingFailed)?;
-        let merchant_secret = connector_meta.merchant_secret;
         let request_data = match item.request.payment_method_data.clone() {
             api::PaymentMethodData::Card(card) => get_card_info(item, &card),
             api::PaymentMethodData::Wallet(wallet) => match wallet {
-                api_models::payments::WalletData::GooglePay(gpay_data) => Ok(Self {
-                    payment_option: PaymentOption {
-                        card: Some(Card {
-                            external_token: Some(ExternalToken {
-                                external_token_provider: ExternalTokenProvider::GooglePay,
-                                mobile_token: gpay_data.tokenization_data.token,
-                            }),
-                            ..Default::default()
-                        }),
-                        ..Default::default()
-                    },
-                    ..Default::default()
-                }),
-                api_models::payments::WalletData::ApplePay(apple_data) => Ok(Self {
-                    payment_option: PaymentOption {
-                        card: Some(Card {
-                            external_token: Some(ExternalToken {
-                                external_token_provider: ExternalTokenProvider::ApplePay,
-                                mobile_token: apple_data.payment_data,
-                            }),
-                            ..Default::default()
-                        }),
-                        ..Default::default()
-                    },
-                    ..Default::default()
-                }),
+                api_models::payments::WalletData::GooglePay(gpay_data) => Ok(Self::from(gpay_data)),
```

Comment:
```suggestion
                api_models::payments::WalletData::GooglePay(gpay_data) => Ok(gpay_data.into()),
```

Prefer into() instead of from().

Reply:
No @jarnura with this it will be easier to navigate to the `From` implementation otherwise we need to search in transformers